### PR TITLE
Automatically create Management object only once

### LINKED ISSUE #175: Management object should be automatically created only once
If initially requested, a management object should be automatically created only once. Object removal should not lead to its recreation.

### PR DESCRIPTION
**Pull request by eromanova**
Closes #175

### CODE REVIEW

File: templates/hmc/values.yaml [hunk 1]
```diff
@@ -6,32 +6,38 @@ admissionWebhook:
   port: 9443
   certDir: "/tmp/k8s-webhook-server/serving-certs/"
 
-controllerManager:
-  manager:
-    args:
-      - --leader-elect
```
Comment (Kshatrix) [inline templates/hmc/values.yaml:12]: we should make it true by default and not expose at all I think

File: internal/helm/values.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	case chartutil.Values:
+		return c, true
+	case map[string]interface{}:
+		return c, true
+	default:
+		return nil, false
+	}
+}
+
+func OverrideValuesWithValues(dst chartutil.Values, src chartutil.Values) {
```
Comment (Kshatrix) [inline internal/helm/values.go:32]: can we use `chartutil.CoalesceTables` instead?

File: internal/controller/release_controller.go [hunk 3]
```diff
@@ -131,28 +132,37 @@ func (p *Poller) ensureManagement(ctx context.Context) error {
...
-				values, err := json.Marshal(release.Config)
-				if err != nil {
-					return err
-				}
-				_ = applyDefaultCoreConfiguration(mgmtObj)
-				mgmtObj.Spec.Core = &hmc.DefaultCoreConfiguration
-				mgmtObj.Spec.Core.HMC.Config = &apiextensionsv1.JSON{
-					Raw: values,
-				}
+				helm.OverrideValuesWithValues(hmcConfig, release.Config)
```
Comment (Kshatrix) [inline internal/controller/release_controller.go:144]:
```suggestion
				chartutil.CoalesceTables(hmcConfig, release.Config)
```

File: internal/controller/release_controller.go [hunk 4]
```diff
@@ -131,28 +132,37 @@ func (p *Poller) ensureManagement(ctx context.Context) error {
...
 			}
 		}
 
+		// Initially set createManagement:false to automatically create Management object only once
+		helm.OverrideValuesWithValues(hmcConfig, map[string]interface{}{
```
Comment (Kshatrix) [inline internal/controller/release_controller.go:149]:
```suggestion
		chartutil.CoalesceTables(hmcConfig, map[string]interface{}{
```

**Review by Kshatrix — CHANGES_REQUESTED** [review]

File: internal/helm/values.go (hunk 2 above)
Comment (eromanova) [reply internal/helm/values.go:32]: Sure. Thank you.

**Review by Kshatrix — APPROVED** [review]

**Pull request merged**